kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/clusternetworkaddressconfigprovider/BrokerAddressPatternUtils.java

```diff
@@ -12,17 +12,20 @@
 import java.util.Set;
 import java.util.TreeSet;
 import java.util.function.Consumer;
+import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 
 import edu.umd.cs.findbugs.annotations.NonNull;
 
 final class BrokerAddressPatternUtils {
     private static final String LITERAL_NODE_ID = "$(nodeId)";
+    private static final String LITERAL_VIRTUAL_CLUSTER_NAME = "$(virtualClusterName)";
     private static final Pattern LITERAL_NODE_ID_PATTERN = Pattern.compile(Pattern.quote(LITERAL_NODE_ID));
     private static final Pattern PORT_SPECIFIER_RE = Pattern.compile("^(.*):([1-9]\\d*)$");
     private static final Pattern TOKEN_RE = Pattern.compile("(\\$\\([^)]+\\))");
-    public static Set<String> EXPECTED_TOKEN_SET = Set.of(LITERAL_NODE_ID);
+    public static Set<String> REQUIRED_TOKEN_SET = Set.of(LITERAL_NODE_ID);
```

Comment:
```suggestion
    public static final Set<String> REQUIRED_TOKEN_SET = Set.of(LITERAL_NODE_ID);
```

Reply:
I shuffled the sets into the specific implementation classes because the accepted set was also used in the port-per-broker implementation, and would have allowed a `$(virtualClusterName)` replacement that would never be replaced. Have added a test to check that it's disallowed for port-per-broker.